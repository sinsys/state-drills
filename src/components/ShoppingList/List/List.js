import React from 'react';
import Card from '../Card/Card';
import './List.css';


class List extends React.Component {

	constructor(props){
		super(props);
		this.state = {
			list: this.props
		}
	}

	render() {
		return (
			<section className="List">
				<header className="List-header">
					<h2>{this.props.title}</h2>
				</header>
				<div className="List-cards">
					{this.props.cards.map((card, index) => {
						if(!card.saved){
							return (
								<Card
									key={this.props.title + "-id-" + card.id + "-index-" + index}
									id={card.id}
									title={card.title}
									content={card.content}
									purchased={false}
									onClickDelete={this.props.onClickDelete}
									onClickSavedCard={this.props.onClickSavedCard}
								/>							
							)
						}
						else return {}
					})}
					<button
						type="button"
						className="List-add-button"
						onClick={() => this.props.onClickAdd(this.props.id)}
					>
						+ Add Card
					</button>
				</div>
			</section>
		)			
	}

}

List.defaultProps = {
  onClickAdd: () => {},
  onClickSavedCard: () => {}
}

export default List;