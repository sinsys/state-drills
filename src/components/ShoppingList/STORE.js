export default {
  lists: [
    {
      id: '1',
      header: 'Daily',
      cardIds: [ 'a', 'b', 'e', 'f', 'g', 'j', 'l', 'm' ],
    },
    {
      id: '2',
      header: 'Weekly',
      cardIds: ['b', 'c', 'd', 'f', 'h', 'i', 'k'],
    },
    {
      id: '3',
      header: 'Monthly',
      cardIds: [ 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm' ],
    },
    {
      id: '4',
      header: 'Seasonal',
      cardIds: [ 'l', 'm' ],
    },
  ],
  allCards: {
    'a': { id: 'a', title: 'First card', content: 'lorem ipsum' },
    'b': { id: 'b', title: 'Second card', content: 'lorem ipsum' },
    'c': { id: 'c', title: 'Third card', content: 'lorem ipsum' },
    'd': { id: 'd', title: 'Fourth card', content: 'lorem ipsum' },
    'e': { id: 'e', title: 'Fifth card', content: 'lorem ipsum' },
    'f': { id: 'f', title: 'Sixth card', content: 'lorem ipsum' },
    'g': { id: 'g', title: 'Seventh card', content: 'lorem ipsum' },
    'h': { id: 'h', title: 'Eighth card', content: 'lorem ipsum' },
    'i': { id: 'i', title: 'Ninth card', content: 'lorem ipsum' },
    'j': { id: 'j', title: 'Tenth card', content: 'lorem ipsum' },
    'k': { id: 'k', title: 'Eleventh card', content: 'lorem ipsum' },
    'l': { id: 'l', title: 'Twelfth card', content: 'lorem ipsum' },
    'm': { id: 'm', title: 'Thirteenth card', content: 'lorem ipsum' },
  }
}