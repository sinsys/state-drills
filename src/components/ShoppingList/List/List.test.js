import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';
import STORE from '../STORE.js';

describe('List component', () => {
	const store = {STORE};
	const props = store.STORE.lists[0];

	it('Renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<List 
				key={"list-" + props.id}
      	title={props.header}
      	cards={props.cardIds.map(card => store.STORE.allCards[card])}
    	/>, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('renders the UI as expected', () => {
		const tree = renderer
			.create(<List 
				key={"list-" + props.id}
      	title={props.header}
      	cards={props.cardIds.map(card => store.STORE.allCards[card])}
    	/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
})