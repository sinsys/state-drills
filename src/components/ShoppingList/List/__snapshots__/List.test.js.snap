// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Daily
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card "
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
      <div
        className="Card-buttons"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Purchased
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Remove
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Save for Later
        </button>
      </div>
    </div>
    <div
      className="Card "
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
      <div
        className="Card-buttons"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Purchased
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Remove
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Save for Later
        </button>
      </div>
    </div>
    <div
      className="Card "
    >
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
      <div
        className="Card-buttons"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Purchased
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Remove
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Save for Later
        </button>
      </div>
    </div>
    <div
      className="Card "
    >
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
      <div
        className="Card-buttons"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Purchased
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Remove
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Save for Later
        </button>
      </div>
    </div>
    <div
      className="Card "
    >
      <h3>
        Seventh card
      </h3>
      <p>
        lorem ipsum
      </p>
      <div
        className="Card-buttons"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Purchased
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Remove
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Save for Later
        </button>
      </div>
    </div>
    <div
      className="Card "
    >
      <h3>
        Tenth card
      </h3>
      <p>
        lorem ipsum
      </p>
      <div
        className="Card-buttons"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Purchased
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Remove
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Save for Later
        </button>
      </div>
    </div>
    <div
      className="Card "
    >
      <h3>
        Twelfth card
      </h3>
      <p>
        lorem ipsum
      </p>
      <div
        className="Card-buttons"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Purchased
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Remove
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Save for Later
        </button>
      </div>
    </div>
    <div
      className="Card "
    >
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
      <div
        className="Card-buttons"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Purchased
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Remove
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Save for Later
        </button>
      </div>
    </div>
    <button
      className="List-add-button"
      onClick={[Function]}
      type="button"
    >
      + Add Card
    </button>
  </div>
</section>
`;
